### ISSUE TITLE
Default settings with a nested hash

### ISSUE THREAD

**Issue by FabianPv** [ledermann/rails-settings#92]
example: 
default settings with this hash 
  **has_settings do |s|
    s.key :conecction, :defaults => {
      :user => '',
      :password  => ''
    }
  end**

it's working perfectly, but if I use a nested hash like this one: 

**has_settings do |s|
    s.key :conecction, :defaults => {
      :user => '',
      :password  => '',
      :credentials => { :token: "", :email:""  }
    }
  end**

once the values are assigned, the database it's only stores the user and the password..

**Comment by olliebennett** [comment]
Yep I'm not sure deeply nested `defaults` are supported. You might need to workaround your problem and consider

```ruby
has_settings do |s|
  s.key :conecction, :defaults => {
    :user => '',
    :password => '',
    :credentials_token => '',
    :credentials_mail => ''
  }
end
```

... although you potentially also want to reconsider the format in which you're storing passwords, if you're storing them in plain text.
